src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -127,9 +127,9 @@ public bool HasValue
         {
             get
             {
-                if (index < 0 || (uint)index >= (uint)Length)
+                if ((uint)index >= (uint)Length)
                 {
-                    throw new IndexOutOfRangeException();
+                    ThrowHelper.ThrowIndexOutOfRangeException();
```

Comment:
Should this rather be an `ArgumentOutOfRangeException`?

Reply:
Yep!